src/main/java/fact/coordinates/EquatorialCoordinate.java

```diff
@@ -0,0 +1,143 @@
+package fact.coordinates;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalUnit;
+
+/**
+ * Created by maxnoe on 22.05.17.
+ *
+ * Represents a celestial coordinate in the equatorial coordinate frame
+ * using right ascension and declination.
+ *
+ * Provides a method to convert to the horizontal coordinate frame
+ */
+public class EquatorialCoordinate implements CelestialCoordinate {
+
+    public final double rightAscensionRad;
+    public final double declinationRad;
+    public static final ZonedDateTime j2000Reference = ZonedDateTime.of(2000, 1, 1, 11, 58, 55, 816000000, ZoneOffset.UTC);
+    public static final double precessionFactorM = 3.075 / 3600 / 12 * Math.PI;
+    public static final double precessionFactorN = 1.336 / 3600 / 12 * Math.PI;
+
+    private EquatorialCoordinate(double rightAscensionRad, double declinationRad) {
+        this.rightAscensionRad = rightAscensionRad;
+        this.declinationRad = declinationRad;
+    }
+
+    public static EquatorialCoordinate fromRad(double rightAscensionRad, double declinationRad) {
+        return new EquatorialCoordinate(rightAscensionRad, declinationRad);
+    }
+
+    public static EquatorialCoordinate fromDegrees(double rightAscensionDeg, double declinationDeg) {
+        return new EquatorialCoordinate(Math.toRadians(rightAscensionDeg), Math.toRadians(declinationDeg));
+    }
+
+    public static EquatorialCoordinate fromHourAngleAndRad(double rightAscensionHA, double declinationRad) {
+        double rightAscensionRad = rightAscensionHA / 12.0 * Math.PI;
+        return new EquatorialCoordinate(rightAscensionRad, declinationRad);
+    }
+
+    public static EquatorialCoordinate fromHourAngleAndDegrees(double rightAscensionHA, double declinationDeg) {
+        double rightAscensionRad = rightAscensionHA / 12.0 * Math.PI;
+        return new EquatorialCoordinate(rightAscensionRad, Math.toRadians(declinationDeg));
+    }
+
+    static double[] calculatePrecessionCorrection(double rightAscensionRad, double declinationRad, ZonedDateTime observationTime) {
```

Comment:
can we add a comment here citing the astronomical almanach. this is whre this comes from right?

Reply:
I will add the reference.